hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/RegionReplicaReplicationEndpoint.java

```diff
@@ -598,6 +598,8 @@ public V call() throws Exception {
    */
   static class RegionReplicaReplayCallable extends
       RegionAdminServiceCallable<ReplicateWALEntryResponse> {
+    private final MetricsSink metricsSink = new MetricsSink();
```

Comment:
hmm, i dont think we want this here, in the callable. it will just be GC'd after the call is completed. Instead take a look at how they increment the `sink.getSkippedEditsCounter()` above.

also, it'd be good to add a jira upstream and we probably should just start with the upstream PR rather than start here. you can mark the PR a draft to avoid any other committers looking at it and we can go back and forth on the review before making it real.

that said, i think region replication has changed a bunch between master/branch-2. so you'll probably need to investigate how to implement for each.

Reply:
Got it, thanks for the feedback, I'll start the work on the upstream PR.